src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -1001,6 +1001,161 @@ class OPENASSETIO_CORE_EXPORT Manager {
                  const BatchElementErrorCallback& errorCallback);
 
+  /**
+   * Register should be used to 'publish' new entities either when
+   * originating new data within the application process, or
+   * referencing some existing file, media or information.
+   *
+   * See documentation for the <!--
+   * --> @ref register_(const EntityReferences&, <!--
+   * --> const trait::TraitsDatas&, const ContextConstPtr&, <!--
+   * --> const RegisterSuccessCallback&, <!--
+   * --> const BatchElementErrorCallback&)
+   * "callback variation" for more details on register_ behaviour.
+   *
+   * Any errors that occur during the register_ call will be immediately
+   * thrown as an exception, either from the @ref manager plugin (for
+   * errors not specific to the entity reference) or as a
+   * @fqref{BatchElementException} "BatchElementException"-derived
+   * error.
+   *
+   * @param entityReference Entity reference to register to.
+   *
+   * @param entityTraitsData The data to register for the entity.
+   *
+   * @param context Context The calling context.
+   *
+   * @param errorPolicyTag  Parameter for selecting the appropriate
+   * overload (tag dispatch idiom). See @ref BatchElementErrorPolicyTag.
+   *
+   * @return Updated reference to use for future interactions with the
+   * resulting new entity.
+   */
+  // NOLINTNEXTLINE(readability-identifier-naming)
+  EntityReference register_(const EntityReference& entityReference,
+                            const TraitsDataPtr& entityTraitsData, const ContextConstPtr& context,
+                            const BatchElementErrorPolicyTag::Exception& errorPolicyTag = {});
+
+  /**
+   * Register should be used to 'publish' new entities either when
+   * originating new data within the application process, or
+   * referencing some existing file, media or information.
+   *
+   * See documentation for the <!--
+   * --> @ref register_(const EntityReferences&, <!--
+   * --> const trait::TraitsDatas&, const ContextConstPtr&, <!--
+   * --> const RegisterSuccessCallback&, <!--
+   * --> const BatchElementErrorCallback&)
+   * "callback variation" for more details on register_ behaviour.
+   *
+   * If successful, the result is populated with an updated reference
+   * for use in future interactions with the resulting new entity
+   *
+   * Otherwise, the result is populated with an error object detailing
+   * the reason for the failure to register this particular entity.
+   *
+   * Errors that are not specific to the entity will be thrown as an
+   * exception.
+   *
+   * @param entityReference Entity reference to register to.
+   *
+   * @param entityTraitsData The data to register for the entity.
+   *
+   * @param context Context The calling context.
+   *
+   * @param errorPolicyTag  Parameter for selecting the appropriate
+   * overload (tag dispatch idiom). See @ref BatchElementErrorPolicyTag.
+   *
+   * @return Updated reference to use for future interactions with the
```

Comment:
https://github.com/OpenAssetIO/OpenAssetIO/pull/879/commits/b1b53934ae2af0caae1e8ae2627088f9129506f1

Reply:
Umm, OK I didn't specifically mean the words "or an error" :laughing: - but I guess it works, since any more does just duplicate what's written above. I think I elaborated a bit in the corresponding `preflight` PR. But if you're happy with it, then it's fine with me.

Reply:
That's what you get for using quotations!

Reply:
Here, took the words directly from the preflight PR : https://github.com/OpenAssetIO/OpenAssetIO/pull/879/commits/feb934fc8b30eb0e4dc19e843ce504fceff339dd

Reply:
:+1: Brilliant, couldn't have worded it any better myself :thinking: